### ISSUE TITLE
Encoding/Decoding issue when a PING request is received with random bytes

### ISSUE THREAD

**Issue by ashmeet-kandhari** [fanout/java-gripcontrol#4]
Whenever a PING request is received from websocket client (in our case python websocket library) the encoder decoder is manipulating the data so finally the ping requests are failing on client side

Note: This works when the PING data is string type.

Sample logs:

```
2021-05-17 21:44:42,398 - DEBUG - client > Frame(fin=True, opcode=9, data=b'\x1b\x88\xf4\xee', rsv1=False, rsv2=False, rsv3=False)
2021-05-17 21:44:42,424 - DEBUG - client - event = data_received(<4 bytes>)
2021-05-17 21:44:42,428 - DEBUG - client < Frame(fin=True, opcode=10, data=b'\x1b\xef', rsv1=False, rsv2=False, rsv3=False)
2021-05-17 21:44:42,432 - DEBUG - client - received unsolicited pong: 1bef
```

**Setup:**
* Pushpin as websocket client
* Websocket over http for to handle requests to pushpin
* Springboot java service with gripcontrol library

**Version:**
* Springboot: 2.4.5
* java-gripcontrol: 1.1.1

**Sample Code:**
```
@PostMapping(value = "/websocket", consumes = "application/websocket-events")
    public ResponseEntity<Object> serverPing(@RequestBody(required = false) byte[] body) {
         List<WebSocketEvent> inEvents = GripControl.decodeWebSocketEvents(body);
         List<WebSocketEvent> outEvents = new ArrayList<>();
         HttpHeaders responseHeaders = new HttpHeaders();

          if (isEventType(inEvents, "OPEN")) {
              ...
        } else if (isEventType(inEvents, "PING")) {
            outEvents.addAll(inEvents);
            // change ping to pong
            outEvents.set(0, new WebSocketEvent("PONG", inEvents.get(0).content));
              responseHeaders.add(HttpHeaders.CONTENT_TYPE, AppConstants.APPLICATION_TYPE_WS);
            responseHeaders.add(AppConstants.EXTENSION_HEADER_NAME, "grip; message-prefix=\"\"");
            
            String resp = GripControl.encodeWebSocketEvents(outEvents);
             return new ResponseEntity<>(
                    resp, responseHeaders, HttpStatus.OK);
}...
```

Client Library Reference: 
https://pypi.org/project/websockets/

**Comment by harmony7** [comment]
I've been looking at this and so far I've been able to reproduce this with your same byte sequence.

The problem is that our data structure at the moment assumes a UTF-8 string payload when a binary one should also be allowed.

I will make time in the next few days to submit a fix.

**Comment by ashmeet-kandhari** [comment]
Thanks @harmony7 ,

Let me know if I can be of any help.

**Comment by harmony7** [comment]
I've released version 1.2.0, which fixes this issue by storing the data in a WebSocketEvent as binary data (`byte[]`) instead of a string (`String`), which would lose data that isn't UTF-8.  See the test `testDecodeWebSocketEventsBinary()`.

Unfortunately, we've had to make one potentially breaking change.. that is, the `content` field on `WebSocketEvent` is now `final`. It usually isn't manipulated anyway after the instance is already created, but it may be a breaking change if code assumes it can be changed. Create a new instance instead of attempting to mutate an existing instance.

**Comment by harmony7** [comment]
One more thing I must mention is that to reencode the WebSocketEvent, please use the new method `encodeWebSocketEventsBinary` (returns `byte[]`) instead of `encodeWebSocketEvents` (returns `String`).
`encodeWebSocketEvents` should only be used for events where the payload is guaranteed to be a UTF-8 string.
